packages/flutter_test/lib/src/finders.dart

```diff
@@ -460,57 +469,544 @@ class CommonFinders {
+
+  @override
+  final SemanticsNode root;
+}
+
+/// Class that wraps an implementation of [FinderBase<Element>] and implements
+/// [Finder] for backwards compatibility with the previous [Finder] API.
+final class _FinderBox implements Finder {
+  _FinderBox(this._unboxed);
+
+  final FinderBase<Element> _unboxed;
+
+  @override
+  Iterable<Element>? _precacheResults;
+
+  @override
+  Iterable<Element> get allCandidates => _unboxed.allCandidates;
+
+  @override
+  Iterable<Element> apply(Iterable<Element> candidates) {
+    return findInCandidates(candidates);
+  }
+
+  @override
+  Finder at(int index) {
+    return _FinderBox(_unboxed.at(index));
+  }
+
+  @override
+  String get description => requirements;
+
+  @override
+  String get requirements => _unboxed.requirements;
+
+  @override
+  Iterable<Element> evaluate() {
+    final Iterable<Element> result = _precacheResults ?? apply(allCandidates);
+    _precacheResults = null;
+    return result;
+  }
+
+  @override
+  Finder get first => _FinderBox(_unboxed.first);
+
+  @override
+  Finder hitTestable({Alignment at = Alignment.center}) {
+    return _HitTestableFinder(_unboxed, at);
+  }
+
+  @override
+  Finder get last => _FinderBox(_unboxed.last);
+
+  @override
+  bool precache() {
+    assert(_precacheResults == null);
+    if (tryFind()) {
+      return true;
+    }
+    _precacheResults = null;
+    return false;
+  }
+
+  @override
+  bool get skipOffstage => !includeOffstage;
+
+  @override
+  bool get includeOffstage {
+    if (_unboxed is ElementFinderMixin) {
+      return (_unboxed as ElementFinderMixin).includeOffstage;
+    } else {
+      return false;
+    }
+  }
+
+  @override
+  FinderResult<Element> find() {
+    return _unboxed.find();
+  }
+
+  @override
+  Iterable<Element> findInCandidates(Iterable<Element> candidates) {
+    return _unboxed.findInCandidates(candidates);
+  }
+
+  @override
+  FinderResult<Element> get found => _unboxed.found;
+
+  @override
+  FinderResult<Element>? get _found {
+    throw UnimplementedError(
+      'This should not be used. '
+      'All methods and properties should be passed through to `_unboxed` or '
+      'reimplemented to match `Finder`.',
+    );
+  }
+  @override
+  set _found(FinderResult<Element>? value) {
+    throw UnimplementedError(
+      'This should not be used. '
+      'All methods and properties should be passed through to `_unboxed` or '
+      'reimplemented to match `Finder`.',
     );
   }
 
-  Iterable<Element>? _cachedResult;
+  @override
+  bool get hasFound => _unboxed.hasFound;
+
+  @override
+  bool tryFind() {
+    return _unboxed.tryFind();
+  }
+
+  @override
+  String toString({bool describeSelf = false}) {
+    final String result = _unboxed.toString(describeSelf: describeSelf);
+    _unboxed._found = null;
+    return result;
+  }
+}
+
+/// Searches a widget tree and returns nodes that match a particular
+/// pattern.
+// TODO(pdblasi-google): Deprecate Finder. @ToBeDeprecated(
```

Comment:
What is left to do to do this deprecation?

Reply:
I actually made a list! #127136